GatewayAuthenticationIdentityIT.getTopologyRequestSucceedsWithValidAuthToken is flaky

**Summary**

> Try to answer the following as best as possible

- How often does the test fail?
- Does it block your work?
- Do we suspect that it is a real failure?

**Failures**

```
Expecting
  <RetriableClientFutureImpl[Failed with the following stack trace:
io.grpc.StatusRuntimeException: CANCELLED
	at io.grpc.Status.asRuntimeException(Status.java:537)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:481)
	at io.grpc.internal.DelayedClientCall$DelayedListener$3.run(DelayedClientCall.java:489)
	at io.grpc.internal.DelayedClientCall$DelayedListener.delayOrExecute(DelayedClientCall.java:453)
	at io.grpc.internal.DelayedClientCall$DelayedListener.onClose(DelayedClientCall.java:486)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:574)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:72)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:742)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:723)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.io.IOException: Failed while requesting access token with status code 401 and message Unauthorized.
	at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsProvider.fetchCredentials(OAuthCredentialsProvider.java:162)
	at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsCache.computeIfMissingOrInvalid(OAuthCredentialsCache.java:100)
	at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsProvider.applyCredentials(OAuthCredentialsProvider.java:84)
	at io.camunda.zeebe.client.impl.ZeebeCallCredentials.lambda$applyRequestMetadata$0(ZeebeCallCredentials.java:48)
	... 3 more
]>
to be completed within 5S.
```

**Logs**

[Test results for [IT] QA Integration Tests (2).zip](https://github.com/camunda/zeebe/files/13510468/Test.results.for.IT.QA.Integration.Tests.2.zip)

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/gateway/GatewayAuthenticationIdentityIT.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/gateway/GatewayAuthenticationIdentityIT.java
@@ -27,16 +27,10 @@ import io.grpc.Metadata;
 import io.grpc.Metadata.Key;
 import io.grpc.Status;
 import io.grpc.StatusRuntimeException;
-import java.net.URI;
-import java.net.http.HttpClient;
-import java.net.http.HttpRequest;
-import java.net.http.HttpResponse;
-import java.net.http.HttpResponse.BodyHandlers;
 import java.time.Duration;
 import java.util.Map;
 import java.util.concurrent.ExecutionException;
 import org.assertj.core.api.InstanceOfAssertFactories;
-import org.awaitility.Awaitility;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
@@ -98,12 +92,15 @@ public class GatewayAuthenticationIdentityIT {
           .withEnv("KEYCLOAK_CLIENTS_0_PERMISSIONS_0_DEFINITION", "write:*")
           .withEnv("IDENTITY_RETRY_ATTEMPTS", "90")
           .withEnv("IDENTITY_RETRY_DELAY_SECONDS", "1")
+          // this will enable readiness checks by spring to await ApplicationRunner completion
+          .withEnv("MANAGEMENT_ENDPOINT_HEALTH_PROBES_ENABLED", "true")
+          .withEnv("MANAGEMENT_HEALTH_READINESSSTATE_ENABLED", "true")
           .withNetwork(NETWORK)
           .withExposedPorts(8080, 8082)
           .waitingFor(
               new HttpWaitStrategy()
                   .forPort(8082)
-                  .forPath("/actuator/health")
+                  .forPath("/actuator/health/readiness")
                   .allowInsecure()
                   .forStatusCode(200))
           .withNetworkAliases("identity");
@@ -118,8 +115,6 @@ public class GatewayAuthenticationIdentityIT {
 
   @BeforeAll
   static void beforeAll() {
-    awaitCamundaRealmAvailabilityOnKeycloak();
-
     ZEEBE
         .withBrokerConfig(
             cfg -> {
@@ -205,29 +200,6 @@ public class GatewayAuthenticationIdentityIT {
       assertThat(client.newTopologyRequest().send().toCompletableFuture())
           .succeedsWithin(Duration.ofSeconds(1));
     }
-  }
-
-  /**
-   * Awaits the presence of the Camunda realm and openid keys on the keycloak container. Once
-   * Keycloak and Identity booted up, Identity will eventually configure the Camunda Realm on
-   * Keycloak.
-   */
-  private static void awaitCamundaRealmAvailabilityOnKeycloak() {
-    final var httpClient = HttpClient.newHttpClient();
-    final HttpRequest request =
-        HttpRequest.newBuilder()
-            .uri(URI.create(getKeycloakRealmAddress() + "/protocol/openid-connect/certs"))
-            .build();
-    Awaitility.await()
-        .atMost(Duration.ofSeconds(120))
-        .pollInterval(Duration.ofSeconds(5))
-        .ignoreExceptions()
-        .untilAsserted(
-            () -> {
-              final HttpResponse<String> response =
-                  httpClient.send(request, BodyHandlers.ofString());
-              assertThat(response.statusCode()).isEqualTo(200);
-            });
   }
 
   private static String getKeycloakRealmAddress() {

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java
@@ -40,11 +40,6 @@ import io.camunda.zeebe.test.util.junit.AutoCloseResources;
 import io.camunda.zeebe.test.util.junit.AutoCloseResources.AutoCloseResource;
 import io.camunda.zeebe.test.util.record.RecordingExporter;
 import io.camunda.zeebe.test.util.testcontainers.ContainerLogsDumper;
-import java.net.URI;
-import java.net.http.HttpClient;
-import java.net.http.HttpRequest;
-import java.net.http.HttpResponse;
-import java.net.http.HttpResponse.BodyHandlers;
 import java.nio.file.Path;
 import java.sql.Connection;
 import java.sql.DriverManager;
@@ -56,7 +51,6 @@ import java.util.Map;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import java.util.function.Supplier;
-import org.awaitility.Awaitility;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
@@ -172,12 +166,15 @@ public class MultiTenancyOverIdentityIT {
           .withEnv("IDENTITY_DATABASE_NAME", DATABASE_NAME)
           .withEnv("IDENTITY_DATABASE_USERNAME", DATABASE_USER)
           .withEnv("IDENTITY_DATABASE_PASSWORD", DATABASE_PASSWORD)
+          // this will enable readiness checks by spring to await ApplicationRunner completion
+          .withEnv("MANAGEMENT_ENDPOINT_HEALTH_PROBES_ENABLED", "true")
+          .withEnv("MANAGEMENT_HEALTH_READINESSSTATE_ENABLED", "true")
           .withNetwork(NETWORK)
           .withExposedPorts(8080, 8082)
           .waitingFor(
               new HttpWaitStrategy()
                   .forPort(8082)
-                  .forPath("/actuator/health")
+                  .forPath("/actuator/health/readiness")
                   .allowInsecure()
                   .forStatusCode(200))
           .withNetworkAliases("identity");
@@ -217,8 +214,6 @@ public class MultiTenancyOverIdentityIT {
         .withRecordingExporter(true)
         .start()
         .await(TestHealthProbe.READY);
-
-    awaitCamundaRealmAvailabilityOnKeycloak();
 
     associateTenantsWithClient(List.of(DEFAULT_TENANT, TENANT_A), ZEEBE_CLIENT_ID_TENANT_A);
     associateTenantsWithClient(List.of(DEFAULT_TENANT, TENANT_B), ZEEBE_CLIENT_ID_TENANT_B);
@@ -1352,32 +1347,6 @@ public class MultiTenancyOverIdentityIT {
               "Expected to handle gRPC request BroadcastSignal with tenant identifier 'tenant-b'")
           .withMessageContaining("but tenant is not authorized to perform this request");
     }
-  }
-
-  /**
-   * Awaits the presence of the Camunda realm and openid keys on the keycloak container. Once
-   * Keycloak and Identity booted up, Identity will eventually configure the Camunda Realm on
-   * Keycloak.
-   */
-  private static void awaitCamundaRealmAvailabilityOnKeycloak() {
-    final var httpClient = HttpClient.newHttpClient();
-    final HttpRequest request =
-        HttpRequest.newBuilder()
-            .uri(
-                URI.create(
-                    "http://localhost:%d%s/protocol/openid-connect/certs"
-                        .formatted(KEYCLOAK.getFirstMappedPort(), KEYCLOAK_PATH_CAMUNDA_REALM)))
-            .build();
-    Awaitility.await()
-        .atMost(Duration.ofSeconds(120))
-        .pollInterval(Duration.ofSeconds(5))
-        .ignoreExceptions()
-        .untilAsserted(
-            () -> {
-              final HttpResponse<String> response =
-                  httpClient.send(request, BodyHandlers.ofString());
-              assertThat(response.statusCode()).isEqualTo(200);
-            });
   }
 
   /**

